<!DOCTYPE html><html lang="en" data-theme="light"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0,viewport-fit=cover"><title>使用GATK进行SNPs和indels的挖掘 | Wu Hou</title><meta name="author" content="Wu Hou"><meta name="copyright" content="Wu Hou"><meta name="format-detection" content="telephone=no"><meta name="theme-color" content="#ffffff"><meta name="description" content="实验记录 使用GATK4进行SNPs和indels的挖掘 本实验的raw data 是编号为SSR6023853-SSR6023865,SSR6023882-SSR6023885的fastq格式数据，测序长度为PE150,参考基因组为my.ref.fa,测序的方式是RNA-Seq。 通过前面的过滤raw data和比对操作，我们可以得每一组测序数据与reference比对生成的sam文件">
<meta property="og:type" content="article">
<meta property="og:title" content="使用GATK进行SNPs和indels的挖掘">
<meta property="og:url" content="http://example.com/2024/12/02/%E4%BD%BF%E7%94%A8GATK%E8%BF%9B%E8%A1%8CSNPs%E5%92%8Cindels%E7%9A%84%E6%8C%96%E6%8E%98/index.html">
<meta property="og:site_name" content="Wu Hou">
<meta property="og:description" content="实验记录 使用GATK4进行SNPs和indels的挖掘 本实验的raw data 是编号为SSR6023853-SSR6023865,SSR6023882-SSR6023885的fastq格式数据，测序长度为PE150,参考基因组为my.ref.fa,测序的方式是RNA-Seq。 通过前面的过滤raw data和比对操作，我们可以得每一组测序数据与reference比对生成的sam文件">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="http://example.com/img/butterfly-icon.png">
<meta property="article:published_time" content="2024-12-02T15:10:16.000Z">
<meta property="article:modified_time" content="2024-12-02T15:11:17.773Z">
<meta property="article:author" content="Wu Hou">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://example.com/img/butterfly-icon.png"><link rel="shortcut icon" href="/img/favicon.png"><link rel="canonical" href="http://example.com/2024/12/02/%E4%BD%BF%E7%94%A8GATK%E8%BF%9B%E8%A1%8CSNPs%E5%92%8Cindels%E7%9A%84%E6%8C%96%E6%8E%98/index.html"><link rel="preconnect" href="//cdn.jsdelivr.net"/><link rel="preconnect" href="//busuanzi.ibruce.info"/><link rel="stylesheet" href="/css/index.css"><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free/css/all.min.css"><script>
    (() => {
      
    const saveToLocal = {
      set: (key, value, ttl) => {
        if (!ttl) return
        const expiry = Date.now() + ttl * 86400000
        localStorage.setItem(key, JSON.stringify({ value, expiry }))
      },
      get: key => {
        const itemStr = localStorage.getItem(key)
        if (!itemStr) return undefined
        const { value, expiry } = JSON.parse(itemStr)
        if (Date.now() > expiry) {
          localStorage.removeItem(key)
          return undefined
        }
        return value
      }
    }

    window.btf = {
      saveToLocal,
      getScript: (url, attr = {}) => new Promise((resolve, reject) => {
        const script = document.createElement('script')
        script.src = url
        script.async = true
        Object.entries(attr).forEach(([key, val]) => script.setAttribute(key, val))
        script.onload = script.onreadystatechange = () => {
          if (!script.readyState || /loaded|complete/.test(script.readyState)) resolve()
        }
        script.onerror = reject
        document.head.appendChild(script)
      }),
      getCSS: (url, id) => new Promise((resolve, reject) => {
        const link = document.createElement('link')
        link.rel = 'stylesheet'
        link.href = url
        if (id) link.id = id
        link.onload = link.onreadystatechange = () => {
          if (!link.readyState || /loaded|complete/.test(link.readyState)) resolve()
        }
        link.onerror = reject
        document.head.appendChild(link)
      }),
      addGlobalFn: (key, fn, name = false, parent = window) => {
        if (!false && key.startsWith('pjax')) return
        const globalFn = parent.globalFn || {}
        globalFn[key] = globalFn[key] || {}
        globalFn[key][name || Object.keys(globalFn[key]).length] = fn
        parent.globalFn = globalFn
      }
    }
  
      
      const activateDarkMode = () => {
        document.documentElement.setAttribute('data-theme', 'dark')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#0d0d0d')
        }
      }
      const activateLightMode = () => {
        document.documentElement.setAttribute('data-theme', 'light')
        if (document.querySelector('meta[name="theme-color"]') !== null) {
          document.querySelector('meta[name="theme-color"]').setAttribute('content', '#ffffff')
        }
      }

      btf.activateDarkMode = activateDarkMode
      btf.activateLightMode = activateLightMode

      const theme = saveToLocal.get('theme')
    
          theme === 'dark' ? activateDarkMode() : theme === 'light' ? activateLightMode() : null
        
      
      const asideStatus = saveToLocal.get('aside-status')
      if (asideStatus !== undefined) {
        document.documentElement.classList.toggle('hide-aside', asideStatus === 'hide')
      }
    
      
    const detectApple = () => {
      if (/iPad|iPhone|iPod|Macintosh/.test(navigator.userAgent)) {
        document.documentElement.classList.add('apple')
      }
    }
    detectApple()
  
    })()
  </script><script>const GLOBAL_CONFIG = {
  root: '/',
  algolia: undefined,
  localSearch: undefined,
  translate: undefined,
  highlight: {"plugin":"highlight.js","highlightCopy":true,"highlightLang":true,"highlightHeightLimit":false,"highlightFullpage":false,"highlightMacStyle":false},
  copy: {
    success: 'Copy Successful',
    error: 'Copy Failed',
    noSupport: 'Browser Not Supported'
  },
  relativeDate: {
    homepage: false,
    post: false
  },
  runtime: '',
  dateSuffix: {
    just: 'Just now',
    min: 'minutes ago',
    hour: 'hours ago',
    day: 'days ago',
    month: 'months ago'
  },
  copyright: undefined,
  lightbox: 'null',
  Snackbar: undefined,
  infinitegrid: {
    js: 'https://cdn.jsdelivr.net/npm/@egjs/infinitegrid/dist/infinitegrid.min.js',
    buttonText: 'Load More'
  },
  isPhotoFigcaption: false,
  islazyload: false,
  isAnchor: false,
  percent: {
    toc: true,
    rightside: false,
  },
  autoDarkmode: false
}</script><script id="config-diff">var GLOBAL_CONFIG_SITE = {
  title: '使用GATK进行SNPs和indels的挖掘',
  isPost: true,
  isHome: false,
  isHighlightShrink: false,
  isToc: true,
  isShuoshuo: false
}</script><meta name="generator" content="Hexo 7.3.0"></head><body><div class="post" id="body-wrap"><header class="post-bg" id="page-header"><nav id="nav"><span id="blog-info"><a class="nav-site-title" href="/"><span class="site-name">Wu Hou</span></a><a class="nav-page-title" href="/"><span class="site-name">使用GATK进行SNPs和indels的挖掘</span></a></span><div id="menus"></div></nav><div id="post-info"><h1 class="post-title">使用GATK进行SNPs和indels的挖掘</h1><div id="post-meta"><div class="meta-firstline"><span class="post-meta-date"><i class="far fa-calendar-alt fa-fw post-meta-icon"></i><span class="post-meta-label">Created</span><time class="post-meta-date-created" datetime="2024-12-02T15:10:16.000Z" title="Created 2024-12-02 23:10:16">2024-12-02</time><span class="post-meta-separator">|</span><i class="fas fa-history fa-fw post-meta-icon"></i><span class="post-meta-label">Updated</span><time class="post-meta-date-updated" datetime="2024-12-02T15:11:17.773Z" title="Updated 2024-12-02 23:11:17">2024-12-02</time></span></div><div class="meta-secondline"><span class="post-meta-separator">|</span><span class="post-meta-pv-cv" id="" data-flag-title=""><i class="far fa-eye fa-fw post-meta-icon"></i><span class="post-meta-label">Post Views:</span><span id="busuanzi_value_page_pv"><i class="fa-solid fa-spinner fa-spin"></i></span></span></div></div></div></header><main class="layout" id="content-inner"><div id="post"><article class="container post-content" id="article-container"><h1 id="实验记录-使用gatk4进行snps和indels的挖掘">实验记录
使用GATK4进行SNPs和indels的挖掘</h1>
<p>本实验的raw data
是编号为<strong>SSR6023853-SSR6023865,SSR6023882-SSR6023885的fastq格式</strong>数据，测序长度为PE150,参考基因组为<strong>my.ref.fa</strong>,测序的方式是RNA-Seq。</p>
<p>通过前面的过滤raw
data和比对操作，我们可以得每一组测序数据与reference比对生成的sam文件，将所有的sam文件转为bam文件之后，首先是对每一个文件进行sort，之后执行merge操作（merge之后的bam也是排序好的，理论上是不需要sort的，但我这里又sort了一遍），此时我们可以得到数量为1的bam文件，它是一个包含了我们所有测序数据的比对结果的二进制文件，之后进行GATK的相关操作来进行变异的挖掘。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools sort -@ 10 merged.bam -o merged_sort.bam</span><br></pre></td></tr></table></figure>
<h2 id="工具的下载">工具的下载</h2>
<p>picard:下载最新版本的<a
target="_blank" rel="noopener" href="http://broadinstitute.github.io/picard/">Picard Tools - By Broad
Institute</a></p>
<p>gatk4:下载最新版本的<a
target="_blank" rel="noopener" href="https://github.com/broadinstitute/gatk/releases">Releases ·
broadinstitute/gatk (github.com)</a></p>
<p>java(最新版的)：在官网下载最新版本就行，注意一定要下x64版本的，实验室服务器的CPU是intel的不是ARM架构的</p>
<h2 id="数据的清理">数据的清理</h2>
<h3 id="标记pcr重复">标记PCR重复</h3>
<p>最新版本的GATK将picard已经纳入其中了，所以我们直接使用GATK命令就可以了</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_">$</span><span class="language-bash">raw_bam = ./merged.bam</span></span><br><span class="line"><span class="meta prompt_">$</span><span class="language-bash">java =  <span class="comment">#这里设置环境变量之后可以不填</span></span></span><br><span class="line"><span class="meta prompt_">$</span><span class="language-bash">gatk = <span class="comment">#这里设置环境变量之后可以不填</span></span></span><br><span class="line">nohup $java $gatk MarkDuplicates -I $raw_bam -O ./merged_markdup.bam -M metrics.txt   --MAX_FILE_HANDLES_FOR_READ_ENDS_MAP 1000 &gt; gatk_mkdup.log 2&gt;&amp;1 &amp; #这一步时间会很长</span><br></pre></td></tr></table></figure>
<p>运行会生成<code>merged_bam_markdup.bam</code>和<code>metrics.txt</code></p>
<p>-M metrics.txt设置输出报告，其中包含有关重复标记过程的统计信息。</p>
<p>标记完成之后就可以构建索引文件了：</p>
<p>使用GATK
callSNPs需要的索引文件有三个：<strong>reference的.fai、reference的.dict、merged_bam_markdup.bam文件的.bai文件</strong>。</p>
<h4 id="对bam构建idx">对bam构建idx</h4>
<p>这一步要在去重之后进行，否则会增加算力负担（最好在全部处理完成得到最终bam之后进行）</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools index merged_markdup.bam</span><br></pre></td></tr></table></figure>
<h4 id="对reference构建idx">对reference构建idx</h4>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_">$</span><span class="language-bash">ref=./my.ref.fa</span></span><br><span class="line"><span class="meta prompt_">$</span><span class="language-bash">java <span class="variable">$gatk</span> CreateSequenceDictionary -R <span class="variable">$ref</span> -O my.ref.dict</span></span><br><span class="line">samtools faidx $ref</span><br></pre></td></tr></table></figure>
<h3 id="对indel附近进行重新比对">对indel附近进行重新比对</h3>
<p>realignment操作</p>
<p>要使用GATK对indel周围进行重新比对，你可以使用GATK工具中的<code>IndelRealigner</code>。这个工具可以根据已知的indel信息重新对reads进行比对，以改善indel区域的比对质量。</p>
<p>创建一个目标坐标数据文件（<code>target intervals</code>）：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk RealignerTargetCreator -R $ref -I merged_markdup.bam -o target.intervals</span><br></pre></td></tr></table></figure>
<p>这个命令会分析去重的BAM文件（<code>merged_bam_markdup.bam</code>），识别其中的indel位置，并生成一个目标坐标数据文件（<code>target intervals</code>）。</p>
<p>对reads进行重新比对：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk IndelRealigner -R $ref -I merged_markdup.bam -targetIntervals target.intervals -o merged_markdup_realigned.bam</span><br></pre></td></tr></table></figure>
<p>这个命令会根据目标坐标数据文件（<code>target intervals</code>）对reads进行重新比对，并输出一个重新比对后的BAM文件（<code>merged_markdup_realigned.bam</code>）。</p>
<p>可选：如果你想生成一个基于实际比对观察到的indel的索引文件，以用于以后的分析步骤，你可以执行以下命令：</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk BamIndexGenerator -I realigned.bam</span><br></pre></td></tr></table></figure>
<p>这个命令会生成一个索引文件，它可以为实际比对观察到的indel提供快速检索。</p>
<p>请注意，上述命令中的参数和文件名可能需要根据你的具体情况进行调整。尤其是<code>-R</code>参数要指定正确的参考基因组文件，<code>-I</code>参数要指定正确的输入BAM文件，以及<code>-o</code>参数要指定输出文件的名称。</p>
<h3
id="碱基质量矫正base-quality-score-recalibration-bqsr">碱基质量矫正Base
quality score recalibration (BQSR)</h3>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk BaseRecalibrator -I merged_markdup_realigned.bam -R $ref --known-sites dbSNP.vcf.gz -O recal_data.table</span><br></pre></td></tr></table></figure>
<p><code>dbSNP.vcf</code>是http://www.ncbi.nlm.nih.gov/SNP/NCBI的公开数据，我们做的植物如果没有的话可以不设置这个参数，不会影响最终结果。</p>
<p>生成的<code>recal_data.table</code>其中记录了参考基因组与样本数据之间的差异，以及需要校正的碱基质量信息。</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="meta prompt_">#</span><span class="language-bash">将BQSR结果进行应用，生成最终的bam文件</span></span><br><span class="line">gatk ApplyBQSR -R $ref -I merged_bam_markdup.bam --bqsr-recal-file recal_data.table -O merged_markdup_realigned_recal.bam</span><br></pre></td></tr></table></figure>
<p>最终过滤完成的文件为<code>merged_markdup_realigned_recal.bam</code></p>
<h2 id="使用gatk进行variant-calling">使用GATK进行Variant Calling</h2>
<p>最新的GATK版本给出的算法为<code>HaplotypeCaller</code>:</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk HaplotypeCaller -R $ref -I merged_markdup_realigned_recal.bam --dbsnp dbSNP.vcf.gz -O raw_variants.vcf</span><br></pre></td></tr></table></figure>
<p>最后得到<code>raw_variants.vcf</code>,
<code>--dbsnp dbSNP.vcf.gz</code>这一项没有公共数据也可以不写。</p>
<h2 id="进行vcf文件的过滤">进行VCF文件的过滤</h2>
<p>在使用GATK进行变异位点调用后，通常需要对原始的VCF文件进行进一步过滤，以过滤掉可能的假阳性（false
positive）变异。</p>
<p>下面是使用GATK提供的工具VariantFiltration来对VCF文件进行过滤的一般步骤：</p>
<ol type="1">
<li><p>创建一个过滤条件（Filter
Expression）：根据你的需求，创建一个过滤条件来过滤掉低质量的变异位点。例如，你可以将一些常见的过滤条件包括：QUAL
&lt; 30.0 (质量值低于30)，DP &lt; 10 (reads深度低于10)，FS &gt; 60.0
(Fisher
Strand值大于60)等等。你可以根据具体的需求和实验数据来自定义这些过滤条件。</p></li>
<li><p>使用<code>VariantFiltration</code>进行过滤：运行<code>VariantFiltration</code>工具，并应用上述定义的过滤条件来过滤VCF文件。</p></li>
</ol>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">gatk VariantFiltration -R $ref -V raw_variants.vcf --filter-expression &quot;过滤条件&quot; --filter-name &quot;过滤名称&quot; -O filtered_variants.vcf</span><br></pre></td></tr></table></figure>
<p>其中，<code>$ref</code>是参考基因组的FASTA文件路径，<code>raw_variants.vcf</code>是HaplotypeCaller生成的原始VCF文件，<code>过滤条件</code>是你定义的过滤条件，<code>过滤名称</code>是为过滤条件定义的名称，<code>filtered_variants.vcf</code>是输出的过滤后的VCF文件。</p>
<ol start="3" type="1">
<li>可选：可以根据你的需要添加或修改过滤条件，并多次运行VariantFiltration命令来进一步优化过滤参数和过滤结果。</li>
</ol>
</article><div class="post-copyright"><div class="post-copyright__author"><span class="post-copyright-meta"><i class="fas fa-circle-user fa-fw"></i>Author: </span><span class="post-copyright-info"><a href="http://example.com">Wu Hou</a></span></div><div class="post-copyright__type"><span class="post-copyright-meta"><i class="fas fa-square-arrow-up-right fa-fw"></i>Link: </span><span class="post-copyright-info"><a href="http://example.com/2024/12/02/%E4%BD%BF%E7%94%A8GATK%E8%BF%9B%E8%A1%8CSNPs%E5%92%8Cindels%E7%9A%84%E6%8C%96%E6%8E%98/">http://example.com/2024/12/02/%E4%BD%BF%E7%94%A8GATK%E8%BF%9B%E8%A1%8CSNPs%E5%92%8Cindels%E7%9A%84%E6%8C%96%E6%8E%98/</a></span></div><div class="post-copyright__notice"><span class="post-copyright-meta"><i class="fas fa-circle-exclamation fa-fw"></i>Copyright Notice: </span><span class="post-copyright-info">All articles on this blog are licensed under <a target="_blank" rel="noopener" href="https://creativecommons.org/licenses/by-nc-sa/4.0/">CC BY-NC-SA 4.0</a> unless otherwise stated.</span></div></div><div class="tag_share"><div class="post-share"><div class="social-share" data-image="/img/butterfly-icon.png" data-sites="facebook,twitter,wechat,weibo,qq"></div><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/css/share.min.css" media="print" onload="this.media='all'"><script src="https://cdn.jsdelivr.net/npm/butterfly-extsrc/sharejs/dist/js/social-share.min.js" defer></script></div></div><nav class="pagination-post" id="pagination"><a class="pagination-related" href="/2024/12/02/%E6%A8%A1%E5%9E%8B%E7%9A%84%E8%AF%84%E4%BC%B0%E4%B8%8E%E9%80%89%E6%8B%A9/" title="模型的评估与选择"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info"><div class="info-1"><div class="info-item-1">Previous</div><div class="info-item-2">模型的评估与选择</div></div><div class="info-2"><div class="info-item-1">模型的评估与选择 经验误差：训练集上出现的误差 泛化误差：测试集上出现的误差 过拟合：学习器过于强调样本规律，而使其泛化能力下降 欠拟合：学习器没能学好样本规律 测试集和训练集的划分 留出法：数据集划分为测试集和训练集两个，并使用分层抽样，即对应每个结果数据集划分比例相同 交叉验证法：将数据集用分层抽样思想划分为\(k\)个子集，每次抽取一个子集当测试集，其余子集当训练集，从而得到\(k\)组测试集和训练集，进行\(k\)次训练，得到的结果取平均值，又称为\(k\)折交叉验证 自助法：数据集中抽样得到测试集，适用于样本量小的数据集，这一方法不会产生测试集和训练集存在无差异而导致的测试无效，理由如下式： \[ \lim_{m \to \infty}(1-\frac{1}{m})^m \to \frac{1}{e}=36.8\% \] TIPS:完整的训练过程是先使用训练集+测试集训练模型，再使用总数据集使用模型，得到结果 模型评价 错误率： \[ E(f;D)=\int_{x \sim D} \prod(f(x) \neq y)p(x)\rm d...</div></div></div></a><a class="pagination-related" href="/2024/12/02/hello-world/" title="Hello World"><div class="cover" style="background: var(--default-bg-color)"></div><div class="info text-right"><div class="info-1"><div class="info-item-1">Next</div><div class="info-item-2">Hello World</div></div><div class="info-2"><div class="info-item-1">Welcome to Hexo! This is your very first post. Check documentation for more info. If you get any problems when using Hexo, you can find the answer in troubleshooting or you can ask me on GitHub. Quick Start Create a new post 1$ hexo new &quot;My New Post&quot; More info: Writing Run server 1$ hexo server More info: Server Generate static files 1$ hexo generate More info: Generating Deploy to remote sites 1$ hexo deploy More info: Deployment </div></div></div></a></nav></div><div class="aside-content" id="aside-content"><div class="card-widget card-info text-center"><div class="avatar-img"><img src="/img/butterfly-icon.png" onerror="this.onerror=null;this.src='/img/friend_404.gif'" alt="avatar"/></div><div class="author-info-name">Wu Hou</div><div class="author-info-description"></div><div class="site-data"><a href="/archives/"><div class="headline">Articles</div><div class="length-num">3</div></a><a href="/tags/"><div class="headline">Tags</div><div class="length-num">1</div></a><a href="/categories/"><div class="headline">Categories</div><div class="length-num">0</div></a></div><a id="card-info-btn" target="_blank" rel="noopener" href="https://github.com/xxxxxx"><i class="fab fa-github"></i><span>Follow Me</span></a></div><div class="card-widget card-announcement"><div class="item-headline"><i class="fas fa-bullhorn fa-shake"></i><span>Announcement</span></div><div class="announcement_content">This is my Blog</div></div><div class="sticky_layout"><div class="card-widget" id="card-toc"><div class="item-headline"><i class="fas fa-stream"></i><span>Contents</span><span class="toc-percentage"></span></div><div class="toc-content"><ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#%E5%AE%9E%E9%AA%8C%E8%AE%B0%E5%BD%95-%E4%BD%BF%E7%94%A8gatk4%E8%BF%9B%E8%A1%8Csnps%E5%92%8Cindels%E7%9A%84%E6%8C%96%E6%8E%98"><span class="toc-number">1.</span> <span class="toc-text">实验记录
使用GATK4进行SNPs和indels的挖掘</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#%E5%B7%A5%E5%85%B7%E7%9A%84%E4%B8%8B%E8%BD%BD"><span class="toc-number">1.1.</span> <span class="toc-text">工具的下载</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E6%95%B0%E6%8D%AE%E7%9A%84%E6%B8%85%E7%90%86"><span class="toc-number">1.2.</span> <span class="toc-text">数据的清理</span></a><ol class="toc-child"><li class="toc-item toc-level-3"><a class="toc-link" href="#%E6%A0%87%E8%AE%B0pcr%E9%87%8D%E5%A4%8D"><span class="toc-number">1.2.1.</span> <span class="toc-text">标记PCR重复</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#%E5%AF%B9bam%E6%9E%84%E5%BB%BAidx"><span class="toc-number">1.2.1.1.</span> <span class="toc-text">对bam构建idx</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#%E5%AF%B9reference%E6%9E%84%E5%BB%BAidx"><span class="toc-number">1.2.1.2.</span> <span class="toc-text">对reference构建idx</span></a></li></ol></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E5%AF%B9indel%E9%99%84%E8%BF%91%E8%BF%9B%E8%A1%8C%E9%87%8D%E6%96%B0%E6%AF%94%E5%AF%B9"><span class="toc-number">1.2.2.</span> <span class="toc-text">对indel附近进行重新比对</span></a></li><li class="toc-item toc-level-3"><a class="toc-link" href="#%E7%A2%B1%E5%9F%BA%E8%B4%A8%E9%87%8F%E7%9F%AB%E6%AD%A3base-quality-score-recalibration-bqsr"><span class="toc-number">1.2.3.</span> <span class="toc-text">碱基质量矫正Base
quality score recalibration (BQSR)</span></a></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E4%BD%BF%E7%94%A8gatk%E8%BF%9B%E8%A1%8Cvariant-calling"><span class="toc-number">1.3.</span> <span class="toc-text">使用GATK进行Variant Calling</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#%E8%BF%9B%E8%A1%8Cvcf%E6%96%87%E4%BB%B6%E7%9A%84%E8%BF%87%E6%BB%A4"><span class="toc-number">1.4.</span> <span class="toc-text">进行VCF文件的过滤</span></a></li></ol></li></ol></div></div><div class="card-widget card-recent-post"><div class="item-headline"><i class="fas fa-history"></i><span>Recent Posts</span></div><div class="aside-list"><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/12/02/%E6%A8%A1%E5%9E%8B%E7%9A%84%E8%AF%84%E4%BC%B0%E4%B8%8E%E9%80%89%E6%8B%A9/" title="模型的评估与选择">模型的评估与选择</a><time datetime="2024-12-02T15:17:18.000Z" title="Created 2024-12-02 23:17:18">2024-12-02</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/12/02/%E4%BD%BF%E7%94%A8GATK%E8%BF%9B%E8%A1%8CSNPs%E5%92%8Cindels%E7%9A%84%E6%8C%96%E6%8E%98/" title="使用GATK进行SNPs和indels的挖掘">使用GATK进行SNPs和indels的挖掘</a><time datetime="2024-12-02T15:10:16.000Z" title="Created 2024-12-02 23:10:16">2024-12-02</time></div></div><div class="aside-list-item no-cover"><div class="content"><a class="title" href="/2024/12/02/hello-world/" title="Hello World">Hello World</a><time datetime="2024-12-02T05:38:49.381Z" title="Created 2024-12-02 13:38:49">2024-12-02</time></div></div></div></div></div></div></main><footer id="footer"><div id="footer-wrap"><div class="copyright">&copy;2019 - 2024 By Wu Hou</div><div class="framework-info"><span>Framework </span><a target="_blank" rel="noopener" href="https://hexo.io">Hexo</a><span class="footer-separator">|</span><span>Theme </span><a target="_blank" rel="noopener" href="https://github.com/jerryc127/hexo-theme-butterfly">Butterfly</a></div></div></footer></div><div id="rightside"><div id="rightside-config-hide"><button id="readmode" type="button" title="Reading Mode"><i class="fas fa-book-open"></i></button><button id="darkmode" type="button" title="Toggle Between Light and Dark Mode"><i class="fas fa-adjust"></i></button><button id="hide-aside-btn" type="button" title="Toggle Between Single-column and Double-column"><i class="fas fa-arrows-alt-h"></i></button></div><div id="rightside-config-show"><button id="rightside-config" type="button" title="Settings"><i class="fas fa-cog fa-spin"></i></button><button class="close" id="mobile-toc-button" type="button" title="Table of Contents"><i class="fas fa-list-ul"></i></button><button id="go-up" type="button" title="Back to Top"><span class="scroll-percent"></span><i class="fas fa-arrow-up"></i></button></div></div><div><script src="/js/utils.js"></script><script src="/js/main.js"></script><div class="js-pjax"></div><script async data-pjax src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"></script></div></body></html>